Allow "zipcode" in addressbatch()

NickCrews opened the pull request:
Currently, the `address()` function expects the keyword argument "zipcode". However, the `addressbatch()` function, if you use the iterable-of-dicts form of argument, expects each dict to have the key "zip". This makes it so users of this library have to use two different schemas, "zip" in one place and "zipcode" in another, depending on if they use `address()` or `addressbatch()`.

This tweaks `addressbatch()` so if you use an iterable-of-dicts, you can use the key "zipcode" instead of "zip". Now in my user code, I only need to have one schema, the one with "zipcode".

This change *doe not* affect `addressbatch()` when you pass in a CSV file. One, I didn't see an easy way to fix this (first thought would be to wrap the file-like-object in a custom file-like-object that modifies the contents on every call to `read()`). Two, I'd imagine the need isn't as serious here, because if you're using a file-like-object then I'd think that you wouldn't even be writing the code that formats that data.

NickCrews committed:
Allow "zipcode" in addressbatch()

Currently, the `address()` function expects the keyword argument "zipcode". However, the `addressbatch()` function, if you use the iterable-of-dicts form of argument, expects each dict to have the key "zip". This makes it so users of this library have to use two different schemas, "zip" in one place and "zipcode" in another, depending on if they use `address()` or `addressbatch()`.

This tweaks `addressbatch()` so if you use an iterable-of-dicts, you can use the key "zipcode" instead of "zip". Now in my user code, I only need to have one schema, the one with "zipcode".

This change *doe not* affect `addressbatch()` when you pass in a CSV file. One, I didn't see an easy way to fix this (first thought would be to wrap the file-like-object in a custom file-like-object that modifies the contents on every call to `read()`). Two, I'd imagine the need isn't as serious here, because if you're using a file-like-object then I'd think that you wouldn't even be writing the code that formats that data.

fitnr commented:
Thanks for catching the bug. I think it would be better to align everything on the `zip` argument. I guess for backward compatibility it would be useful to also accept "zipcode" in `address()`.

NickCrews commented:
OK, that actually would probably make more sense, I agree.

NickCrews committed:
Make address() be consistent with addressbatch()

This actually is the better way to make it consistent:

- Now the CSV method is actually consistent too.
- It lines up better with the Census API terminology.

NickCrews committed:
Fix logic bug

Both None -> same logic, result set to None
zipcode is used, but not zip -> warning is raised, but zipcode value is still used, preserving old behavior
zipcode and zip are both used -> ValueError is raised, because we don't know which to use
zip is used, but not zipcode -> this is desired new behavior, zip value is used

NickCrews committed:
Fix docs and undo some whitespace munging

NickCrews commented:
I'm doing all this as armchair coding from the GitHub web UI, so do a double check I'm not putting any bugs in here.

fitnr commented:
There was a small bug, so I've added them to a new pull request with an update.  I have a few other small things to tweak, then I'll release a new version. Thanks for raising this.
